neurodamus/cell_distributor.py

```diff
@@ -437,74 +437,84 @@ class LFPManager:
     def __init__(self):
         self._lfp_file = None
 
-    def load_lfp_config(self, lfp_weights_file, circuit_list):
+    def load_lfp_config(self, lfp_weights_file, population_list):
         """Loads lfp weigths from h5 file
         """
         logging.info("Reading LFP configuration info from '%s'", lfp_weights_file)
         import h5py
         try:
             self._lfp_file = h5py.File(lfp_weights_file, 'r')
         except IOError as e:
-            raise ConfigurationError(e)
-        # Check that the file contains the required groups and attributes
-        required_groups = ['electrodes', 'neuron_ids', 'sec_ids']
-        for group in required_groups:
-            if group not in self._lfp_file:
-                raise ConfigurationError("LFP weights file format is wrong.Required group "
-                                         "not found in file: {}".format(group))
-        nodeids_group = self._lfp_file["neuron_ids"]
-        try:
-            circuit = nodeids_group.attrs['circuit']
-        except KeyError:
-            raise ConfigurationError("'circuit' attribute not found in 'neuron_ids' "
-                                     "group of LFP weights file")
-        logging.info("Circuit of the lfp config file: '%s'", circuit)
-        # Check that the circuit matches with any cell manager
-        for circuit_file in circuit_list:
-            circuit_file_is_set = circuit_file != "start.ncs"
-            if not circuit_file_is_set or circuit_file == circuit:
-                break
-        else:
-            logging.warning("Circuits don't match, aborting lfp config reading...")
-            self._lfp_file.close()
-
-    def read_lfp_factors(self, gid, section_ids):
+            raise ConfigurationError(f"Error opening LFP electrodes file: {e}")
+
+        # Check that the file contains the required groups for at least 1 population
+        populations_found = []
+        for pop_name in population_list:
+
+            req_groups = [f'/electrodes/{pop_name}/scaling_factors', f'{pop_name}/node_ids',
+                          f'{pop_name}/offsets']
+            if all(group in self._lfp_file for group in req_groups):
+                populations_found.append(pop_name)
+
+        if not populations_found:
+            raise ConfigurationError("The LFP weights file does not contain the necessary datasets "
+                                     "'scaling_factors', 'node_ids' and 'offsets' "
+                                     "in any of the populations {}.".format(population_list))
+
+    def get_sonata_node_id(self, gid, population_info):
+        return population_info[0], gid - population_info[1] - 1
+
+    def get_node_id_subsets(self, node_id, section_index, num_sections, population_name):
+        node_ids = self._lfp_file[population_name]["node_ids"]
+        # Look for the index of the node_id
+        index = numpy.where(numpy.array(node_ids) == node_id)[0][0]
```

Comment:
`node_ids` is not a numpy array already? Also, If the array is sorted maybe we can use `searchsorted`

Reply:
node_ids is actually a <class 'h5py._hl.dataset.Dataset'> since it is coming from the h5 file.
We don't know for sure if the node_ids are sorted, since the electrodes_file is created from a report, and this report node_ids are not guaranteed to be sorted.